webapp/src/extensionsBrowser.tsx

```diff
@@ -408,71 +405,148 @@ export const ExtensionsBrowser = (props: ExtensionsProps) => {
-                {lf("Extensions")}
-            </div>
-            <div className="header-right"></div>
-        </MenuBar>
-        <div className="extensionSearchHeader">
-            <div className="header">{(lf("Do more with your micro:bit"))}</div>
-            <SearchInput searchHandler={setSearchFor} />
-            <div className="extensionTags">
-                {categoryNames.map(c =>
-                    <div className={"extensionTag " + (selectedTag == c ? "selected" : "")} onClick={() => handleCategoryClick(c)}>{c}</div>
-                )}
-            </div>
-            {/* TODO bring in the import modal in later! <div className="importButton">
-                <span>{lf("or ")}</span>
-                <div className="importButtonLink" onClick={() => setShowImportExtensionDialog(true)}>{lf("import extension")}</div>
-            </div> */}
-        </div>
-        {displayMode == ExtensionView.Search &&
-            <div className="extension-display">
-                <div className="breadcrumbs">
-                    <span className="link" onClick={handleHomeButtonClick}>{lf("Home")}</span>
-                </div>
-                <div className="extension-grid">
-                    {extensionsToShow?.map(scr =>
-                        <ExtensionCard scr={scr} onCardClick={installExtension} learnMoreUrl={scr.fullName ? `/pkg/${scr.fullName}` : undefined}
-                            name={scr.name} imageUrl={scr.imageUrl} description={scr.description} loading={scr.loading} />)}
-                </div>
-            </div>}
-        {displayMode == ExtensionView.Tags &&
-            <div className="extension-display">
-                <div className="breadcrumbs">
-                    <span className="link" onClick={handleHomeButtonClick}>{lf("Home")}</span>
-                    <span>/</span>
-                    <span>{selectedTag}</span>
+    let loadingCardId = 0
+    return (
+        <Modal
+            title={lf("Extensions")}
+            fullscreen={true}
+            className={"extensions-browser"}
+            onClose={props.hideExtensions}
+            helpUrl={"/extensions"}
+        >
+            <div className="ui">
+                {showImportExtensionDialog ? <ImportModal onCancelClick={() => setShowImportExtensionDialog(false)} onImportClick={handleImportUrl} /> : undefined}
+                {deletionCandidate ? <DeleteConfirmationModal ns={deletionCandidate.name} onCancelClick={() => { setDeletionCandidate(undefined) }} onDeleteClick={() => { removeDepAsync(deletionCandidate) }} /> : undefined}
+                <div className="extension-search-header">
+                    <div className="header">{(lf("Do more with your micro:bit"))}</div>
+                    <SearchInput searchHandler={setSearchFor} key={"search-input"}/>
+                    <div className="extension-tags">
+                        {categoryNames.map(c =>
+                            <Button title={lf(c)}
```

Comment:
instead of lf, use rlf. lf only works with static strings, rlf will do a lookup against the list of known strings.